### DISCUSSION TITLE
Defining a Semantic Versioning Policy

### DISCUSSION THREAD

**Discussion by Belco90** [testing-library/eslint-plugin-testing-library#343] [Ideas]
Hey there!

I think we need to define a Semantic Versioning Policy so:
- users of the plugin know what to expect when a new version gets released
- contributors know the impact of every sort of change, so it's easier to coordinate and organize version bumps

I think [Semantic Versioning Policy from ESLint](https://github.com/eslint/eslint#semantic-versioning-policy) itself is the best option and we should stick to that. The idea then would be to mirror it in our README and CONTRIBUTING docs, just changing `eslint:recommended` by our available Shareable Configs.

What do you think? @gndelia @timdeschryver @MichaelDeBoey

**Comment by MichaelDeBoey** [comment]
Following ESLints versioning policy sounds a good idea. 👍

I've searched for @typescript-eslint's versioning, but I couldn't find one.
Since @typescript-eslint is coming from the @eslint team, I assume they're using the same versioning policy.

So The two big repos we should look at are having the same policy and so should we imo.

**Reply by MichaelDeBoey** [reply]
On the other hand: we should be careful to not keep adding extra options just to not make breaking changes.

**Comment by gndelia** [comment]
The ESLint versioning policy sounds great. I would've expected any change to `eslint:recommended` would be a breaking change but their distinction makes sense

**Comment by Belco90** [comment]
Alright, I'll add the Semantic Versioning Policy next week. My idea is to put it in its own file under `docs/` and reference it within `README.md`. Are we happy with this?

**Comment by MichaelDeBoey** [comment]
Taking a closer look at how @typescript-eslint is handling this: they don't seem to follow the ESLint versioning policy.

- [`v4.20.0`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md#4200-2021-03-29) is reporting more
- [`v4.19.0`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md#4190-2021-03-22) is reporting more
- [`v4.13.0`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md#4130-2021-01-11) is reporting more
- ...

Although they sometimes are:
- [`v4.16.0`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md#4160-2021-03-01) is adding an option to the `member-delimiter-style` rule

I'll create an issue on their repo in order to ask what versioning policy they're using, so we can take that into account when deciding what way we go.

**Reply by MichaelDeBoey** [reply]
https://github.com/typescript-eslint/typescript-eslint/issues/3320

**Reply by MichaelDeBoey** [reply]
@bradzacher was so kind to answer my question: https://github.com/typescript-eslint/typescript-eslint/issues/3320#issuecomment-827684800

**Comment by Belco90** [comment]
Merged in https://github.com/testing-library/eslint-plugin-testing-library/pull/387

Thanks for your help!
